axon-cosmosdb/src/main/java/org/axonframework/extensions/cosmosdb/eventsourcing/tokenstore/CosmosTokenStore.java

```diff
@@ -0,0 +1,516 @@
+                            new CosmosItemRequestOptions().setIfMatchETag(current.getETag())
+                    );
+                })
+                .flatMap(r -> {
+                    if (r.getStatusCode() == 200) {
+                        return getCurrent(processorName, segment);
+                    } else {
+                        return Mono.empty();
+                    }
+                })
+                .block();
+        if (response == null || response.getItem() == null) {
+            throw new UnableToClaimTokenException(NO_CURRENT_ITEM_MESSAGE);
+        }
+        return toTrackingToken(response.getItem());
+    }
+
+    @Override
+    public void extendClaim(@Nonnull String processorName, int segment) throws UnableToClaimTokenException {
+        try {
+            getCurrent(processorName, segment)
+                    .flatMap(current -> {
+                        String owner = current.getItem().getOwner();
+                        if (!nodeId.equals(owner)) {
+                            return Mono.error(new UnableToClaimTokenException(
+                                    String.format(CANT_EXTEND_MESSAGE_FORMAT, owner, nodeId)));
+                        }
+                        return getContainer(processorName).replaceItem(
+                                current.getItem().extend(),
+                                current.getItem().getId(),
+                                new PartitionKey(current.getItem().getId()),
+                                new CosmosItemRequestOptions().setIfMatchETag(current.getETag())
+                        );
+                    })
+                    .block();
+        } catch (UnableToClaimTokenException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new UnableToClaimTokenException("Could not extend token claim.", e);
+        }
+    }
+
+    @Override
+    public void releaseClaim(@Nonnull String processorName, int segment) {
+        getCurrent(processorName, segment)
+                .flatMap(current -> {
+                    validateCurrent(current.getItem());
+                    return getContainer(processorName).replaceItem(
+                            current.getItem().release(),
+                            current.getItem().getId(),
+                            new PartitionKey(current.getItem().getId()),
+                            new CosmosItemRequestOptions().setIfMatchETag(current.getETag())
+                    );
+                })
+                .block();
+    }
+
+    @Override
+    public void initializeSegment(@Nullable TrackingToken token, @Nonnull String processorName, int segment)
+            throws UnableToInitializeTokenException {
+        try {
+            getContainer(processorName).createItem(
+                    createInitialItem(token, segment),
+                    new PartitionKey(Integer.toString(segment)),
+                    new CosmosItemRequestOptions().setIfMatchETag(null)
```

Comment:
Curious: Is it even necessary to consciously match with a `null` etag?

Reply:
No, create item will throw if the id already exists. I made some other simplifications as well.